AirSeaFluxCode/src/hum_subs.py

```diff
@@ -353,48 +354,48 @@ def get_hum(hum, T, sst, P, qmeth):
     ----------
     hum : array
         humidity input switch 2x1 [x, values] default is relative humidity
-            x='rh' : relative humidity in %
-            x='q' : specific humidity (kg/kg)
-            x='Td' : dew point temperature (K)
+            x='rh' : relative humidity [%]
+            x='q' : specific humidity [g/kg]
+            x='Td' : dew point temperature [K]
     T : float
-        air temperature in K
+        air temperature [K]
     sst : float
-        sea surface temperature in K
+        sea surface temperature [K]
     P : float
-        air pressure at sea level in hPa
+        air pressure at sea level [hPa]
     qmeth : str
         method to calculate specific humidity from vapor pressure
 
     Returns
     -------
     qair : float
-        specific humidity of air
+        specific humidity of air [g/kg]
     qsea : float
-        specific humidity over sea surface
+        specific humidity over sea surface [g/kg]
 
     """
-    if (hum[0] not in ['rh', 'q', 'Td', 'no']):
-        sys.exit("unknown humidity input")
-        qair, qsea = np.nan, np.nan
-    elif ((hum[0] == 'rh') or (hum[0] == 'no')):
+    if ((hum[0] == 'rh') or (hum[0] == 'no')):
         RH = hum[1]
         if np.all(RH < 1):
-            sys.exit("input relative humidity units should be %")
-            qair, qsea = np.nan, np.nan
-        qsea = qsat_sea(sst, P, qmeth)/1000    # surface water q (kg/kg)
-        qair = qsat_air(T, P, RH, qmeth)/1000  # q of air (kg/kg)
+            raise ValueError("input relative humidity units should be %")
+        qsea = qsat_sea(sst, P, qmeth)  # surface water q [g/kg]
+        qair = qsat_air(T, P, RH, qmeth)  # q of air [g/kg]
     elif hum[0] == 'q':
-        qair = hum[1]
-        qsea = qsat_sea(sst, P, qmeth)/1000  # surface water q (kg/kg)
+        qair = hum[1]  # [g/kg]
+        if np.all(qair < 1):
```

Comment:
Under the most extreme temperatures (colder than -20ish C), a correct value can have q < 1 g/kg. Fairly sure, -20 degC is very unusual for a marine environment, but this is something to watch out.

Look at:
https://www.lenntech.com/calculators/humidity/relative-humidity.htm
and try put some really cold temperatures in.